### PR TITLE
fix: login modal validation

### PR DESCRIPTION
**Pull request by ayusharma**
**Type:** Refactor

The following has been addressed in the PR:

*  There is a related issue? N/A
*  Unit or Functional tests are included in the PR.

**Description:**
- Adds form validation on login modal

### CODE REVIEW

File: src/webui/components/Login/index.js [hunk 1]
```diff
@@ -32,20 +51,30 @@ export default class LoginModal extends Component {
...
     });
   }
 
-  async submitCredentials(event) {
-    // prevents default submit behaviour
+  /**
+   * Clears the username and password field.
+   */
+  handleReset() {
+    this.formRef.current.resetFields();
```
Comment (juanpicado) [inline src/webui/components/Login/index.js]: Here, imho I'd rather reset only the password, username after auth fails should remains.

Comment (juanpicado) [reply src/webui/components/Login/index.js]: @priscilawebdev any thought here?

File: src/webui/components/Login/index.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import React, {Component} from 'react';
+import React, {Component, createRef} from 'react';
```
Comment (juanpicado) [inline src/webui/components/Login/index.js]: 💯 I just learned something new 🙂 `createRef` that's new in React 16.3.

File: src/webui/components/Login/index.js [hunk 3]
```diff
@@ -32,20 +51,30 @@ export default class LoginModal extends Component {
    * set login modal's username and password to current state
    * Required by: <LoginModal />
    */
-  setCredentials(name, e) {
+  setCredentials(key, value) {
     this.setState({
-      [name]: e
+      form: Object.assign({}, this.state.form, {[key]: value})
```
Comment (priscilawebdev) [inline src/webui/components/Login/index.js]: @ayusharma  why don't you use spread ...? I believe it's clearer.

File: src/webui/components/Login/index.js (hunk 1 above)
Comment (priscilawebdev) [reply src/webui/components/Login/index.js]: @juanpicado @ayusharma I just checked how it works in Gmail :) It's as Juan said ... I agree with him.

File: src/webui/components/Login/index.js (hunk 3 above)
Comment (ayusharma) [reply src/webui/components/Login/index.js]: noted 🗒

File: src/webui/components/Login/index.js (hunk 2 above)
Comment (ayusharma) [reply src/webui/components/Login/index.js]: Yes. :)

**Pull request merged**